api/v1alpha2/awsmachine_conversion.go

```diff
@@ -36,13 +36,14 @@ func (src *AWSMachine) ConvertTo(dstRaw conversion.Hub) error { // nolint
 	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil || !ok {
 		return err
 	}
-
 	restoreAWSMachineSpec(&restored.Spec, &dst.Spec)
+	restoreAWSMachineStatus(&restored.Status, &dst.Status)
 
 	return nil
 }
 
 func restoreAWSMachineSpec(restored *infrav1alpha3.AWSMachineSpec, dst *infrav1alpha3.AWSMachineSpec) {
+	dst.FailureDomain = restored.FailureDomain
```

Comment:
Shouldn't this be migrated to/from AvailabilityZone in v1alpha2?

Reply:
Yes! Found the issue we had with the current conversion!